### ISSUE TITLE
from_arrow() fails when arrow array is of type `DataType(null)`

### ISSUE THREAD

**Issue by drahnreb** [scikit-hep/awkward#828]
As everything can be nullable in Arrow (and Parquet) an Arrow Array of type `DataType(null)` with `num_buffers = 1` could be handled separately in `.from_arrow()` or should be excepted.

```python
>>> import awkward as ak
>>> ak.__version__
1.2.2
>>> ak.from_arrow(pa.Table.from_pydict({'null col': [None]}))
AssertionError:
[...]
awkward/operations/convert.py in popbuffers(array, tpe, buffers)
   2755 
   2756         elif isinstance(tpe, pyarrow.lib.DataType):
-> 2757             assert tpe.num_buffers == 2
[...]
```
The problem:
```python
>>> pa.Table.from_pydict({'null col': [None]}).column(0).chunks[0].type.num_buffers
1
```

In the reverse, I experienced that a nullable (valid) awkward Array gets transformed into Record Arrow Array of type (consider nullability on both levels):
```
pyarrow.Table
nullable_field: list<item: double> not null
  child 0, item: double not null
```
That fails to be written to parquet (similar to #678). Hard to reproduce but I suspect a similar or common(?) unhandled case for nullable fields?

**Comment by jpivarski** [comment]
The first issue is fixed in #829; see the test. I simply hadn't known about Arrow's `DataType(null)` and I'm glad the assertion caught it before it became a more confusing error later on. The appropriate translation of this type would be `?unknown`.

As for the second, I'm not sure what the symptom is. Here are some arrays with nullability on one or two levels, converted into StructArrays and Tables—are any of these different from what you expect?

```python
>>> ak.to_arrow(ak.Array([{"x": 1}, {"x": None}, {"x": 3}]))
<pyarrow.lib.StructArray object at 0x7fd9202439a0>
-- is_valid: all not null
-- child 0 type: int64
  [
    1,
    null,
    3
  ]
>>> ak.to_arrow(ak.Array([{"x": 1}, {"x": None}, None, {"x": 3}]))
<pyarrow.lib.StructArray object at 0x7fd920243a00>
-- is_valid:
  [
    true,
    true,
    false,
    true
  ]
-- child 0 type: int64
  [
    1,
    null,
    null,
    3
  ]
>>> ak.to_arrow_table(ak.Array([{"x": 1}, {"x": None}, {"x": 3}]))
pyarrow.Table
x: int64
>>> ak.to_arrow_table(ak.Array([{"x": 1}, {"x": None}, None, {"x": 3}]))
pyarrow.Table
: struct<x: int64>
  child 0, x: int64
```

**Comment by drahnreb** [comment]
Thanks I saw. Great! Incredibly fast again.

I need to investigate the second part a bit more (I was fiddling on low-level and built a "valid" awkward Array but was unable to save it via Arrow due to invalid Arrow/Parquet fields).

As tests passed and your fix works perfectly, I close this issue.
